apps/root/src/pages/transfer/components/recipient-address/components/contact-list-modal/index.tsx

```diff
@@ -0,0 +1,134 @@
+            />
+          </Typography>
+        </Grid>
+        <Grid item sx={{ maxWidth: '350px !important' }} xs={12}>
+          <Button variant="contained" onClick={onOpenAddContactForm} fullWidth>
+            <Typography variant="body1" fontWeight="bold">
+              <FormattedMessage description="addContact" defaultMessage="Add Contact" />
+            </Typography>
+          </Button>
+        </Grid>
+      </Grid>
+    ),
+    [onOpenAddContactForm, themeMode]
+  );
+
+  const noContactsOnSearch = React.useMemo(
+    () => (
+      <Grid item direction="column" gap={2} textAlign="center" color={colors[themeMode].typography.typo3}>
+        <Typography variant="h3">🔍</Typography>
+        <Typography variant="body1" fontWeight="bold">
+          <FormattedMessage description="noContactsFound" defaultMessage="No contacts were found" />
+        </Typography>
+      </Grid>
+    ),
+    [themeMode]
+  );
+
+  const filteredContacts = React.useMemo(
+    () =>
+      contactList.filter(
+        (contact) =>
+          contact.address.toLowerCase().includes(searchValue.toLowerCase()) ||
+          contact.label?.toLowerCase().includes(searchValue.toLowerCase())
+      ),
+    [contactList, searchValue]
+  );
+
+  return (
+    <Modal
+      open={shouldShow}
+      onClose={() => setShouldShow(false)}
+      closeOnBackdrop={true}
+      title={<FormattedMessage description="contactList" defaultMessage="Contact list" />}
+      headerButton={
+        contactList.length !== 0 && (
+          <Button variant="outlined" color="primary" onClick={onOpenAddContactForm}>
+            <Typography variant="body1" fontWeight="bold">
+              <FormattedMessage description="addContact" defaultMessage="Add Contact" />
+            </Typography>
+          </Button>
+        )
+      }
+      maxWidth="sm"
+    >
+      <Grid container direction="column" rowSpacing={6}>
+        <Grid item>
+          <TextField
+            fullWidth
+            placeholder={intl.formatMessage(
+              defineMessage({
+                defaultMessage: 'Search by Alias or Address',
+                description: 'searchContact',
+              })
+            )}
+            value={searchValue}
+            onChange={(e) => setSearchValue(e.target.value)}
+            InputProps={{
+              startAdornment: (
+                <InputAdornment position="start">
+                  <SearchIcon htmlColor={colors[themeMode].typography.typo4} />
+                </InputAdornment>
+              ),
+            }}
+          />
+        </Grid>
+        <Grid item>
+          <Divider sx={{ borderColor: colors[themeMode].border.border2 }} />
+        </Grid>
+        <Grid
+          item
+          container
+          direction="column"
+          rowGap={1}
+          height={(theme) => theme.spacing(65)}
```

Comment:
why are we defining height here? this should take the max height available right?

Reply:
I thought of this when seeing the gradient in the bottom of the Modal, so it may be scrollable. We are implementing that in the future so I have removed the `height` property and still looks fine :D